gradle-jdks-setup/src/main/resources/gradle-jdks-setup.sh

```diff
@@ -132,16 +135,29 @@ for dir in "$APP_GRADLE_DIR"/jdks/*/; do
     echo "Distribution $distribution_url already exists in $jdk_installation_directory"
   else
     # Download and extract the distribution into a temporary directory
-    echo "Distribution $distribution_url does not exist, installing in progress ..."
+    echo "Distribution $jdk_installation_directory does not exist, installing $distribution_url in progress ..."
     in_progress_dir="$tmp_work_dir/$distribution_local_path.in-progress"
     mkdir -p "$in_progress_dir"
     cd "$in_progress_dir"
     if command -v curl > /dev/null 2>&1; then
       echo "Using curl to download $distribution_url"
-      curl -C - "$distribution_url" | tar -xzf -
+      #
+      if echo "$distribution_url" | grep -q ".zip"; then
```

Comment:
Can probably use a `case` statement to avoid using `grep`:

```
case "$distribution_url" in
  *.zip)
    # ...
    ;;
  *)
    # ...
    ;;
esac
```